.github/workflows/forum.yml

```
@@ -0,0 +1,79 @@
+name: Move to Typst Forum
+on:
+  issues:
+    types:
+      - labeled
+jobs:
+  add-comment:
+    if: github.event.label.name == 'to-forum'
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+    steps:
+      - name: Call Discourse API to create forum post
+        env:
+          ISSUE_BODY: ${{ github.event.issue.body }}
+          ISSUE_TITLE: ${{ github.event.issue.title }}
+          ISSUE_NUMBER: ${{ github.event.issue.number }}
+          ISSUE_URL: ${{ github.event.issue.html_url }}
+          ISSUE_USER_LOGIN: ${{ github.event.issue.user.login }}
+          REPOSITORY_OWNER: ${{ github.repository_owner }}
+        run: |
+          RAW="This topic was moved from GitHub issue ${ISSUE_NUMBER}. Please continue the conversation here.
+
+          Here is the original issue:
+          [quote=\"${ISSUE_USER_LOGIN}\"]
+          ${ISSUE_BODY}
+          [/quote]
+
+          Browse the previous discussion at ${ISSUE_URL}"
```

Did you know you could use shell HEREDOC syntax here?

---

I did not! How would that look? I have to admit, the only thing that [this explanation](https://stackoverflow.com/a/1655389/1567835) did is slightly confuse me.

---

The explanation you linked to is for another layer of complexity, that OP did *not* want to expand things in the content of the message. You do, which is why I thought the HEREDOC would be a clearer way to express that such that quotes don't need to be escaped and finding the ending quote mark isn't a game of needle in the haystack:

```suggestion
          read -d '' RAW << END_OF_BODY
          This topic was moved from GitHub issue ${ISSUE_NUMBER}. Please continue the conversation here.

          Here is the original issue:
          [quote="${ISSUE_USER_LOGIN}"]
          ${ISSUE_BODY}
          [/quote]

          Browse the previous discussion at ${ISSUE_URL}
          END_OF_BODY
```

I think the default shell for this runner is already `bash`, but if not you can set it with `shell: bash` just before this `run:` key.

---

Thank you! `bash` is the default, [see here](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#defaultsrunshell).